src/atlas_f/field/atlas_FieldSet_module.fypp

```diff
@@ -293,6 +348,322 @@ subroutine halo_exchange(this,on_device)
   call atlas__FieldSet__halo_exchange(this%CPTR_PGIBUG_A, on_device_int)
 end subroutine
 
+!-------------------------------------------------------------------------------
+
+subroutine set_host_needs_update_value(this, value)
+  use atlas_field_c_binding
+  use atlas_Field_module, only: atlas_Field
+  class(atlas_FieldSet), intent(inout) :: this
+  logical, intent(in), optional :: value
+  type(atlas_Field) :: field
+  integer :: i, value_int
```

Comment:
Please modify `integer :: ...` to `integer(c_int) :: ...` everywhere and add `use, intrinsic :: iso_c_binding, only: c_int` where needed